web/packages/teleport/src/services/bot/bot.ts

```diff
@@ -36,8 +37,16 @@ import {
   FlatBot,
 } from './types';
 
-export function createBot(config: CreateBotRequest): Promise<void> {
-  return api.post(cfg.getBotsUrl(), config);
+export function createBot(
+  config: CreateBotRequest,
+  mfaResponse: MfaChallengeResponse
```

Comment:
Here the `mfaResponse` arg is required whereas in `createBotToken` it's optional. Is there a reason for that? Seeing how currently both functions are used in the same way, I'd expect them both to use a similar approach to that arg.

Reply:
thanks for the catch, it should be optional since normally there's a error catch later on that will ask user for mfaResponse

Reply:
so normally, you can call `createBot` without passing a mfaResponse, because underneath we use [fetchJsonWithMfaAuthnRetry](https://github.com/gravitational/teleport/blob/e6331a779a58cb4c04bfe70dccc7a702a73adbbe/web/packages/teleport/src/services/api/api.ts#L149), which if mfa is required, it asks for re-authn, and then retries the original call

but if you want to call multiple apis that require re-authn, instead of each call asking for re-authn (like it was done previous to my PR), we can ask the user to re-authn previous to these calls and pass in the mfaResponse we already received (it can be re-used, lasts 5 minutes) 

hope that made sense 😅

Reply:
Ah yeah that makes sense. It'd be good to add a comment with what you wrote to `fetchJsonWithMfaAuthnRetry`. Otherwise I don't know if we ever mention that reusing MFA responses is a thing.